core/src/replay_stage.rs

```diff
@@ -79,7 +79,7 @@ impl Drop for Finalizer {
 pub struct ReplayStageConfig {
     pub my_pubkey: Pubkey,
     pub vote_account: Pubkey,
-    pub voting_keypair: Option<Arc<Keypair>>,
+    pub voting_keypair: Option<Arc<Mutex<Box<dyn Signer>>>>,
```

Comment:
How about?

```rust
ReplayStageConfig<T: Signer> {
   pub voting_keypair: Option<Arc<T>>
}
```

That might need to be `Signer + SomeOtherTraits`

Reply:
Thanks, I think that might work. I'll try that direction and rework. If successful, it should make this whole implementation cleaner. I'll also play around with removing the Arc around LedgerWallet